docs/spec.yml

```diff
@@ -304,6 +303,260 @@ paths:
+        - in: path
+          name: createdTimestamp
+          schema:
+            type: string
+          required: true
+          description: Date at which the record was created. Sort key of the DynamoDB table.
+      operationId: letter
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object 
+              required:
+                - vtmUsername
+                - recipientEmailAddress
+                - letterType
+                - paragraphId
+              properties:
+                vtmUsername:
+                  type: string
+                recipientEmailAddress:
+                  type: string
+                letterType:
+                  type: string
+                  enum:
+                    - 'trailer acceptance'
+                    - 'trailer rejection'
+                paragraphId:
+                  type: number
+                  enum:
+                   - 3
+                   - 4
+                   - 5
+                   - 6
+                   - 7
+      responses:
+        '200':
+          description: Success
+        '401':
+          description: Not authenticated
+        '403':
+          description: Unauthorised Access
+        '404':
+          description: Operation not supported
+  '/v3/technical-records/promote/{systemNumber}/{createdTimestamp}':
+    post:
+      summary: 'Promotes the specified technical record to current'
+      tags:
+        - Promote a technical record
+      parameters:
+        - in: path
+          name: systemNumber
+          schema:
+            type: string
+          required: true
+          description: This represents the systemNumber of the vehicle. Partition key of the DynamoDB table.
+        - in: path
+          name: createdTimestamp
+          schema:
+            type: string
+          required: true
+          description: Date at which the record was created. Sort key of the DynamoDB table.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object 
+              required:
+                - reasonForPromoting
+              properties:
+                reasonForPromoting:
+                  type: string
+      responses:
+        '200':
+          description: Returns the promoted technical record
+          content:
+            application/json:
+              schema:
+                $ref: "https://raw.githubusercontent.com/dvsa/cvs-type-definitions/v6.4.0/json-schemas/v3/tech-record/get/psv/complete/index.json"
+        '400':
+          description: Bad request
+        '401':
+          description: Not authenticated
+        '403':
+          description: Unauthorised Access
+        '404':
+          description: Operation not supported
+        '422':
+          description: Unprocessable request
+        '500':
+          description: Validation Error
```

Comment:
Validation errors are usually 400 requests

Reply:
Yep, there are otehr places where validation causes an actual error to be thrown, but not here. Agree that validation errors should be a 400, and we should probably revise the other endpoints